### DISCUSSION TITLE
Resize screenshot image

### DISCUSSION THREAD

**Discussion by Youdiie** [bevyengine/bevy#9333] [Q&A]
Hello, after 0.11 version, I found screenshot feature, but I couldn't change the size of a screenshot result png.
I can change bevy window size, but I just want to resize image.
Is there any way to change size of screenshot image?
Any hint is welcome!

**Comment by mockersf** [comment]
The method [`save_screenshot_to_disk`](https://docs.rs/bevy/latest/bevy/render/view/window/screenshot/struct.ScreenshotManager.html#method.save_screenshot_to_disk) does not let you resize, however you can use [`take_screenshot`](https://docs.rs/bevy/latest/bevy/render/view/window/screenshot/struct.ScreenshotManager.html#method.take_screenshot) instead where you are free to do any modification you want on the image before saving it, for example with [`resize`](https://docs.rs/bevy/latest/bevy/render/texture/struct.Image.html#method.resize).

Once you have changed the image, you can reuse the code from `save_screenshot_to_disk` on how to save it (most of the complexity of that function is for wasm support): https://docs.rs/bevy_render/latest/src/bevy_render/view/window/screenshot.rs.html#62-120

```rust
    screenshot_manager.take_screenshot(window, move |img| {
            // You can modify `img` here to resize it
            match img.try_into_dynamic() {
                Ok(dyn_img) => {
                    // Or modify `dyn_img` here for more powerful methods
                    match image::ImageFormat::from_path(&path) {
                        Ok(format) => {
                            let img = dyn_img.to_rgb8();
                            match img.save_with_format(&path, format) {
                                Ok(_) => info!("Screenshot saved to {}", path.display()),
                                Err(e) => error!("Cannot save screenshot, IO error: {e}"),
                            }
                        }
                        Err(e) => {
                            error!("Cannot save screenshot, requested format not recognized: {e}")
                        }
                    }
                }
                Err(e) => error!("Cannot save screenshot, screen format cannot be understood: {e}"),
            }
        });
```

**Reply by Youdiie** [reply]
@mockersf Thanks for answering! With your code, I could resize dyn_img and save it :)
However, I think modifying 'dyn_img' is resizing the result of changing the texture into an image. The result quality of `dyn_img.resize(5000,3000,filter)` was not very good..
Is there any way to set size before changing the texture into an image? 🙏

**Reply by mockersf** [reply]
With that resolution, are you trying to scale up the image?

The `image` crate that is used has a a few different filters, but I'm not sure how well they work... you can try them all in that example: https://github.com/image-rs/image/tree/master/examples/scaleup

Resizing before taking the screenshot is currently not possible, they are tied to a specific window. It should be possible to target a camera instead, and then add an offscreen camera of the size you want, but that would require a PR to Bevy before it's available.

**Reply by Youdiie** [reply]
I understand. I wish it can be possible to resize before taking the screenshot in future version.
Thanks for kind explanation!
